satpy/tests/reader_tests/modis_tests/_modis_fixtures.py

```diff
@@ -251,8 +254,11 @@ def create_hdfeos_test_file(filename: str,
     if include_metadata:
         if geo_resolution is None or file_shortname is None:
             raise ValueError("'geo_resolution' and 'file_shortname' are required when including metadata.")
+        elif geo_resolution == -999 or geo_resolution == -9999:
+            setattr(h, 'StructMetadata.0', _create_struct_metadata_cmg(geo_resolution))  # noqa
+        else:
+            setattr(h, 'StructMetadata.0', _create_struct_metadata(geo_resolution))  # noqa
```

Comment:
How about extract this whole set of geo-related things to a `_add_geo_metadata` and calling it with `(h, geo_resolution)`?

Reply:
Done!